### PR TITLE
NSB.Testing - Add override to upgrade guide

### PR DESCRIPTION
**Pull request by mikeminutillo**
Upgrade guide changes for https://github.com/Particular/NServiceBus.Testing/pull/121

### CODE REVIEW

File: nservicebus/upgrades/testing-6to7.md [hunk 1]
```diff
@@ -31,4 +31,12 @@ Note how the `ExpectSagaCompleted` and `ExpectSagaNotCompleted` expectations mus
...
 ## ExpectHandleCurrentMessageLater
 
-The `ExpectHandleCurrentMessageLater` has been obsoleted as `IMessageHandlerContext.HandleCurrentMessageLater()` has been deprecated in NServiceBus Version 7.
\ No newline at end of file
+The `ExpectHandleCurrentMessageLater` has been obsoleted as `IMessageHandlerContext.HandleCurrentMessageLater()` has been deprecated in NServiceBus Version 7.
+
+
+## WhenHandling
+
+An override of the `WhenHandling` method has been added which accepts a preconstructed message.
```
Comment (timbussmann) [inline nservicebus/upgrades/testing-6to7.md]: wouldn't that be a overload?

File: Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿namespace Testing_6.UpgradeGuides._6to7
```
Comment (timbussmann) [inline Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: where is this snippet used?

Comment (weralabaj) [inline Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: remove namespace

File: Snippets/Testing/Testing_7/UpgradeGuides/6to7/WhenHandling.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿namespace Testing_7.UpgradeGuides._6to7
```
Comment (weralabaj) [inline Snippets/Testing/Testing_7/UpgradeGuides/6to7/WhenHandling.cs]: remove namespace

File: Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs (hunk 2 above)
Comment (mikeminutillo) [reply Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: @timbussmann it's line 41 of the upgrade guide doc. It shows that the old method created the message for you but the new overload allows you to send an existing message

Comment (mikeminutillo) [reply Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: @weralabaj happy to do this but all of the other classes in this solution have namespaces that match the folder. I can submit a separate PR to remove them all?

File: nservicebus/upgrades/testing-6to7.md (hunk 1 above)
Comment (mikeminutillo) [reply nservicebus/upgrades/testing-6to7.md]: :+1:

File: Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs (hunk 2 above)
Comment (timbussmann) [reply Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: so it shows both versions on the same page?

Comment (mikeminutillo) [reply Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: I think so. I'm pretty sure that's what the `AssertSagaCompletionIs` snippet does on the page.

image

Comment (weralabaj) [reply Snippets/Testing/Testing_6/UpgradeGuides/6to7/WhenHandling.cs]: OK, let's ignore for now then and we'll clean up separately

**Pull request merged**